connectionpool.go

```diff
@@ -508,6 +541,14 @@ func (pool *hostConnPool) connect() (err error) {
 		return nil
 	}
 
+	pool.scyllaHostInfo.updateFromConn(conn)
+	if usesShardAwarePort && shardHint != conn.scyllaSupported.shard {
+		// Something's wrong, we got a connection to the wrong shard.
+		// It may be caused by network configuration issues (e.g. NAT
+		// modifies the source port).
+		return errors.New("gocql: connected to a wrong shard; make sure that the source port is not translated by NAT")
```

Comment:
I'm not sure what to do in this case. If there is a NAT that consistently changes the source port so that a wrong shard keeps being assigned, the driver will get stuck in a loop where it keeps opening new connections.

One idea I have is to temporarily disable using shard-aware port for new connections if misconfigured NAT is detected. This way the client will not get stuck in a loop, and it will give opportunity to switch to the better method when NAT gets fixed.

Is it a good idea? How long should the timeout be?

Reply:
I added a hardcoded one-minute timeout. I also added a test which verifies that this fallback behavior works.